example_test.go

```diff
@@ -67,42 +155,69 @@ func NewMux(lc fx.Lifecycle, logger *log.Logger) *http.ServeMux {
 	return mux
 }
 
+// Register mounts our HTTP handler on the mux.
+//
+// Register is a typical top-level application function: it takes a generic
+// type like ServeMux, which typically comes from a third-party library, and
+// introduces it to a type that contains our application logic. In this case,
+// that introduction consists of registering an HTTP handler. Other typical
+// examples include registering RPC procedures and starting queue consumers.
+//
+// Fx calls these functions invocations, and they're treated differently from
+// the constructor functions above. Their arguments are still supplied via
+// dependency injection and they may still return an error to indicate
+// failure, but any other return values are ignored.
+//
+// Unlike constructors, invocations are called eagerly. See the main function
+// below for details.
 func Register(mux *http.ServeMux, h http.Handler) {
 	mux.Handle("/", h)
 }
 
 func Example() {
 	app := fx.New(
-		// Provide all the constructors we need.
-		fx.Provide(NewLogger, NewHandler, NewMux),
-		// Before starting, register the handler. This forces resolution of all
-		// the types Register function depends on: *http.ServeMux and http.Handler.
-		// Since the mux is now being used, its startup hook gets registered
-		// and the application includes an HTTP server.
+		// Provide all the constructors we need, which teaches Fx how we'd like to
+		// construct the *log.Logger, http.Handler, and *http.ServeMux types.
+		// Remember that constructors are called lazily, so this block doesn't do
+		// anything on its own.
```

Comment:
optional: "anything on its own (aside from providing the constructors into the application)"

Reply:
Reworded to "doesn't do much".